File: try_java/src/main/java/com/example/model/UserRepository.java
package com.example.model;

import javax.validation.constraints.NotBlank;
import java.util.List;
import java.util.Optional;

public interface UserRepository {
    Optional<UserDto> findById(long id);
    UserDto save(@NotBlank String name);
//    UserDto saveWithException(@NotBlank String name);
    List<UserDto> findAll();
}

File: try_kotlin/src/main/kotlin/com/example/model/UserRepository.kt
package com.example.model

import javax.validation.constraints.NotBlank

interface UserRepository {
    fun findById(id: Long): UserDto?
    fun save(name: @NotBlank String): UserDto?

    //    UserDto saveWithException(@NotBlank String name);
    fun findAll(): List<UserDto>?
}